src/openrct2-ui/windows/EditorObjectSelection.cpp

```diff
@@ -222,1149 +215,1315 @@ static constexpr const int32_t window_editor_object_selection_animation_divisor[
     {
-        _listSortType = RIDE_SORT_TYPE;
-        _listSortDescending = false;
+        auto tab = selected_tab;
+        return static_cast<ObjectType>(tab);
     }
-    else
+
+    /**
+     * Takes the y coordinate of the clicked on scroll list
+     * and converts this into an object selection.
+     * Returns the position in the list.
+     * Object_selection_flags, installed_entry also populated
+     *
+     * rct2: 0x006AA703
+     */
+    int32_t GetObjectFromObjectSelection(ObjectType object_type, int32_t y)
     {
-        _listSortType = RIDE_SORT_RIDE;
-        _listSortDescending = false;
+        int32_t listItemIndex = y / SCROLLABLE_ROW_HEIGHT;
+        if (listItemIndex < 0 || static_cast<size_t>(listItemIndex) >= _listItems.size())
+            return -1;
+
+        return listItemIndex;
     }
 
-    VisibleListRefresh(w);
-    w->Invalidate();
-}
+    void SetPressedTab()
+    {
+        for (size_t i = 0; i < std::size(ObjectSelectionPages); i++)
+        {
+            pressed_widgets &= ~(1 << (WIDX_TAB_1 + i));
+        }
+        pressed_widgets |= 1LL << (WIDX_TAB_1 + selected_tab);
+    }
 
-static void WindowEditorObjectSelectionSetPressedTab(rct_window* w)
-{
-    for (size_t i = 0; i < std::size(ObjectSelectionPages); i++)
+    /**
+     *
+     * rct2: 0x006D33E2
+     */
+    void ManageTracks()
     {
-        w->pressed_widgets &= ~(1 << (WIDX_TAB_1 + i));
+        set_every_ride_type_invented();
+        set_every_ride_entry_invented();
+
+        gEditorStep = EditorStep::DesignsManager;
+
+        int32_t entry_index = 0;
+        for (; object_entry_get_chunk(ObjectType::Ride, entry_index) == nullptr; entry_index++)
+            ;
+
+        rct_ride_entry* ride_entry = get_ride_entry(entry_index);
+        uint8_t ride_type = ride_entry_get_first_non_null_ride_type(ride_entry);
+
+        auto intent = Intent(WC_TRACK_DESIGN_LIST);
+        intent.putExtra(INTENT_EXTRA_RIDE_TYPE, ride_type);
+        intent.putExtra(INTENT_EXTRA_RIDE_ENTRY_INDEX, entry_index);
+        context_open_intent(&intent);
     }
-    w->pressed_widgets |= 1LL << (WIDX_TAB_1 + w->selected_tab);
-}
+};
 
 /**
- * Takes the y coordinate of the clicked on scroll list
- * and converts this into an object selection.
- * Returns the position in the list.
- * Object_selection_flags, installed_entry also populated
  *
- *  rct2: 0x006AA703
+ * rct2: 0x006AA64E
  */
-static int32_t GetObjectFromObjectSelection(ObjectType object_type, int32_t y)
+rct_window* WindowEditorObjectSelectionOpen()
 {
-    int32_t listItemIndex = y / SCROLLABLE_ROW_HEIGHT;
-    if (listItemIndex < 0 || static_cast<size_t>(listItemIndex) >= _listItems.size())
-        return -1;
-
-    return listItemIndex;
+    auto* window = window_bring_to_front_by_class(WC_EDITOR_OBJECT_SELECTION);
+    if (window == nullptr)
+    {
+        window = WindowCreate<EditorObjectSelectionWindow>(WC_EDITOR_OBJECT_SELECTION, 755, 400, WF_10 | WF_RESIZABLE);
```

Comment:
Pass in `WF_CENTRE_SCREEN` to match the legacy `WindowCreateCentred`.

```
WF_10 | WF_RESIZABLE | WF_CENTRE_SCREEN
```

Also you can use `WindowFocusOrCreate` to do the focus part as well simplifying this function even further.

Reply:
Thank you. Done https://github.com/OpenRCT2/OpenRCT2/pull/16027/commits/627b4f78a43e1179c6086feb48195e7f486242cb